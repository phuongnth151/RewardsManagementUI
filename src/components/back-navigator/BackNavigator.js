import React, {Component} from 'react';
import backBtn from '../../images/backBtn.png'
import './BackNavigation.css';



class BackNavigator extends Component{

    constructor(props){
        super(props);
        this.onBackClick = this.onBackClick.bind(this);
    }

    onBackClick() {
        console.log("back clicked!");
        this.props.setItemQuantity("");
        this.props.history.replace({pathname: './'});
        
    };

    render(){
        return(
          <div>
              <div className="backTemp" onClick={this.onBackClick}>
                  <img className='backBtn' src={backBtn} alt="back-button"/>
                  <div className="backBtnTxt">Cancel</div>
              </div>
          </div>
        );
    }
}

export default BackNavigator;