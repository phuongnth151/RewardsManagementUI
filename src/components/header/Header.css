.header{
    color: white;
    background-color: #2B3134;
    width: 100%;
    z-index: 1;
    line-height: 2.8em;
    font-size: 18px;
    text-align: center;
}

.skuNotFoundDiv{
    color: white;
    background-color: #ED1C24;
    height: 55px;
    z-index: 1;
    position: relative;
    width: 100%;
    display: none;
}

.skuNotFoundText{
    width: 250px;
    margin: 0 auto;
    font-size: 14px;
    margin-top: 12px;
    float: left;
}

.skuNotFoundCloseBtn{
    float: left;
    width: 30px;
    margin-top: 17px;
    margin-left: 20px;
    font-size: 18px;
}

.skuNotFoundIconDiv{
    float: left;
    width: 50px;
}

.skuNotFoundIcon{
    width: 30px;
    height: 30px;
    margin-left: 10px;
    margin-top: 10px;
    float: left;
}

.titleTxt {
    margin-top: 16px;
    font-size: 18px;
    display: block;
    font-family: sans-serif;
    font-weight: 500;
    text-align: center;
}