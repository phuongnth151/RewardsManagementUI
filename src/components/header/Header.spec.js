import React from 'react'
import {shallow,mount} from 'enzyme'
import Header from './Header'
import BackNavigator from "../back-navigator/BackNavigator";

let component;

it('Make sure <Header/> component is getting rendered without any issues', () => {

    component = shallow(<Header><div className="headerTxt">Inventory Prep</div> </Header>)
    expect(component).toMatchSnapshot();
} )


it('Make sure <Header/> component has black background color', () => {

    component = shallow(<Header />)
    console.log(component.find(".header").exists())
    expect(component.find(".header").exists())
} )

it('Make sure <Header/> is able to get the children text and pass it to the header', () => {

    component = shallow(<Header><div className="headerTxt">Inventory Prep</div></Header>)
    expect(component.find(".headerTxt").text()).toEqual("Inventory Prep")
} )

it('Make sure <Header/> is able to get the cancel text', () => {

    component = shallow(<Header><BackNavigator /><div className="headerTxt">Inventory Prep</div></Header>)
    expect(component.find("BackNavigator").exists())
    expect(component.find(".headerTxt").text()).toEqual("Inventory Prep")
} )


