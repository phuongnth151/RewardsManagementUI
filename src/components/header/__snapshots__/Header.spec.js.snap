// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure <Header/> component is getting rendered without any issues 1`] = `ShallowWrapper {}`;
